Add HydroPowerSimulations Docs

rodrigomha opened the pull request.

kdayday commented on docs/src/index.md:
```diff
@@ -6,8 +6,8 @@ CurrentModule = HydroPowerSimulations
 ## Overview
 
-`HydroPowerSimulations.jl` is a [`Julia`](http://www.julialang.org) package that provides blah blah
+`HydroPowerSimulations.jl` is a [`Julia`](http://www.julialang.org) package that extends [`PowerSimulations.jl`](https://github.com/NREL-Sienna/PowerSimulations.jl) for modeling of hydro generation technology in production cost modeling simulations.
```
```suggestion
`HydroPowerSimulations.jl` is a [`Julia`](http://www.julialang.org) package that extends [`PowerSimulations.jl`](https://nrel-sienna.github.io/PowerSimulations.jl/stable/) for modeling of hydro generation technology in production cost modeling simulations.
```

jd-lara replied:
`production cost modeling simulations` -> `operational simulations`

kdayday replied:
@rodrigomha, can you update the hyperlink above, too?

rodrigomha replied:
Miss that one

kdayday commented on docs/src/tutorials/single_stage_model.md:
Review the compiled html to remove all errors -- looks like PSCB isn't in the environment, and all code examples are failing

kdayday commented on docs/src/tutorials/single_stage_model.md:
```diff
@@ -0,0 +1,62 @@
+# [Operation Problem with `HydroPowerSimulations.jl`](@id op_problem)
+
+**Originally Contributed by:** Rodrigo Henriquez-Auba
```
I would remove

kdayday commented on docs/src/tutorials/single_stage_model.md:
```diff
@@ -0,0 +1,62 @@
+!!! note
+    
+    `PowerSystemCaseBuilder.jl` is a helper library that makes it easier to reproduce examples in the documentation and tutorials. Normally you would pass your local files to create the system data instead of calling the function `build_system`.
+    For more details visit [PowerSystemCaseBuilder Documentation](https://nrel-sienna.github.io/PowerSystems.jl/stable/tutorials/powersystembuilder/)
```
Link is failing due to PSY re-org. Either update, or preferably, set up DocumenterInterlinks.jl to PSY and use: `[PowerSystemCaseBuilder Documentation](@extref powersystembuilder)`

kdayday commented on docs/src/tutorials/single_stage_model.md:
````diff
@@ -0,0 +1,62 @@
+sys = build_system(PSITestSystems, "c_sys5_hy")
+```
+
+With a single `HydroDispatch`:
````
Nice job breaking the code up into chunks. Please preface each call to a new function with a hyperlink to its docstring, e.g., 
```
With a single [`PowerSystems.HydroDispatch`](@extref)
```

kdayday commented on docs/src/tutorials/single_stage_model.md:
```diff
@@ -0,0 +1,62 @@
+
+## Decision Model
+
+Setting up the formulations, including hydro using `HydroDispatchRunOfRiver`
```
Maybe break this up to clarify what is coming standard from PSI, and reiterating the hydro part is from this extension package?

jd-lara replied:
This one is defined here in HSS. The only formulation from PSI that's not defined here is `FixedOutput`

kdayday commented on docs/src/tutorials/single_stage_model.md:
````diff
@@ -0,0 +1,62 @@
+set_device_model!(template, HydroDispatch, HydroDispatchRunOfRiver)
+```
+
+```@repl op_problem
````
Needs a text intro with hyperlinks

kdayday commented on src/core/expressions.jl:
```diff
@@ -1,4 +1,12 @@
+"""
+Expression for HydroPumpedStorage that keep track
```
Please turn this and all following references to HydroPumpedStorage into hyperlinks to the docstring using `@extref`: http://juliadocs.org/DocumenterInterLinks.jl/stable/

kdayday commented on src/feedforwards.jl:
```diff
@@ -1,12 +1,23 @@
-    source - From where the data comes
-    affected_values -
+  - `component_type::Type{<:PSY.Component}` : Specify the type of component on which the Feedforward will be applied
+  - `source::Type{T}` : Specify the VariableType, ParameterType or AuxVariableType as the source of values for the Feedforward
```
Add hyperlinks for all Types named in args lists

kdayday commented on src/hydrogeneration_constructor.jl:
```diff
@@ -1419,6 +1419,9 @@ function PSI.construct_device!(
 end
 
+"""
+Construct model for HydroGen with HydroCommitmentRunOfRiver Formulation
```
Hyperlinks

kdayday reviewed (changes requested):
Nice! I've added some comments based to improve formatting -- see best practices on [general formatting](https://nrel-sienna.github.io/InfrastructureSystems.jl/dev/docs_best_practices/how-to/general_formatting/#hyperlinks). I think the extension packages would benefit from using [DocumenterInterlinks.jl](http://juliadocs.org/DocumenterInterLinks.jl/stable/) to other packages. Let me know if you need help. Thanks!

jd-lara reviewed (changes requested):
This looks good to me. The only needed thing is to address @kdayday's comments

rodrigomha commented:
Thanks for the comments: Here is the preview with the latest changes: https://nrel-sienna.github.io/HydroPowerSimulations.jl/previews/PR51/

@kdayday I did my best to add as much as possible for refs end extrefs. Hopefully I did not miss anything. I also realized that we do not have docstrings for `ActivePowerBalance` in PSI, so we will need to add those when we do the PSI docs update.

kdayday commented on docs/src/tutorials/single_stage_model.md:
````diff
@@ -0,0 +1,71 @@
+res = OptimizationProblemResults(model)
+```
+
+with dispatch variable results for the hydro:
````
```suggestion
Use [`read_variable`](@ref InfrastructureSystems.Optimization.read_variable) to read in the dispatch variable results for the hydro:
```

kdayday reviewed (commented):
@rodrigomha Looks good, -- I left 2 more comments/suggestions.

This closes #49, #50, correct? And any more updates to #38? 

I also just added a new issue #52 to follow up on migrating the formulation library into the api -- please add any feedback there, thanks!

rodrigomha commented:
> @rodrigomha Looks good, -- I left 2 more comments/suggestions.
> 
> This closes #49, #50, correct? And any more updates to #38?
> 
> I also just added a new issue #52 to follow up on migrating the formulation library into the api -- please add any feedback there, thanks!

I think this should also close #38, and yeah #50. Regarding #49, I'm not sure if this new HydroReservoir, if Pedro's new Reservoir model that we have not implemented yet.

kdayday commented:
> > @rodrigomha Looks good, -- I left 2 more comments/suggestions.
> > This closes #49, #50, correct? And any more updates to #38?
> > I also just added a new issue #52 to follow up on migrating the formulation library into the api -- please add any feedback there, thanks!
> 
> I think this should also close #38, and yeah #50. Regarding #49, I'm not sure if this new HydroReservoir, is Pedro's new Reservoir model that we have not implemented yet.

Sounds good. Can you also update the formatter to close #38? Number 4 here: https://nrel-sienna.github.io/InfrastructureSystems.jl/dev/docs_best_practices/how-to/requirements_checklist/#For-Existing-Packages Thanks!

rodrigomha commented:
> > > @rodrigomha Looks good, -- I left 2 more comments/suggestions.
> > > This closes #49, #50, correct? And any more updates to #38?
> > > I also just added a new issue #52 to follow up on migrating the formulation library into the api -- please add any feedback there, thanks!
> > 
> > 
> > I think this should also close #38, and yeah #50. Regarding #49, I'm not sure if this new HydroReservoir, is Pedro's new Reservoir model that we have not implemented yet.
> 
> Sounds good. Can you also update the formatter to close #38? Number 4 here: https://nrel-sienna.github.io/InfrastructureSystems.jl/dev/docs_best_practices/how-to/requirements_checklist/#For-Existing-Packages Thanks!

Thanks Kate. I updated to be like what we have in IS.
